src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -803,6 +803,9 @@ public FormValidation doValidate(@QueryParameter String userName,
             site.setTimeout(timeout);            
             try {
                 JiraSession session = site.createSession();
+                if (session == null) {
+                    return FormValidation.error("Failed to login to JIRA");
```

Comment:
Won't that hide the actual reason it failed to login (because of not showing the stacktrace) as lines 812 and 815 should handle those cases and additionally provide a stacktrace to the output logs with the underlying cause?

Reply:
JiraSite.createSession() might return null session when username or password is not provided. Throwing NullpointerException seems not that helpful to end user but developers.
OK, I'd revert this before a better solution.